src/bytes.ts

```diff
@@ -34,34 +35,62 @@ export const setLengthLeft = function(msg: any, length: number, right: boolean =
     return msg.slice(-length)
   }
 }
-export const setLength = setLengthLeft
 
 /**
- * Right Pads an `Array` or `Buffer` with leading zeros till it has `length` bytes.
+ * Right Pads a `Buffer` with leading zeros till it has `length` bytes.
  * Or it truncates the beginning if it exceeds.
- * @param msg the value to pad (Buffer|Array)
+ * @param msg the value to pad (Buffer)
  * @param length the number of bytes the output should be
- * @return (Buffer|Array)
+ * @return (Buffer)
  */
-export const setLengthRight = function(msg: any, length: number) {
-  return setLength(msg, length, true)
+export const setLengthRight = function(msg: Buffer, length: number) {
+  return setLengthLeft(msg, length, true)
 }
 
 /**
- * Trims leading zeros from a `Buffer` or an `Array`.
+ * Trims leading zeros from a `Buffer`.
+ * @param a (Buffer)
+ * @return (Buffer)
+ */
+export const unpadBuffer = function(a: any): Buffer {
+  assertIsBuffer(a)
+  return stripZeros(a) as Buffer
+}
+
+/**
+ * Trims leading zeros from a `Array` (of numbers).
+ * @param a (number[])
+ * @return (number[])
+ */
+export const unpadArray = function(a: number[]): number[] {
+  assertIsArray(a)
+  return stripZeros(a) as number[]
+}
+
+/**
+ * Trims leading zeros from a hex-prefixed `String`.
+ * @param a (String)
+ * @return (String)
+ */
+export const unpadHexString = function(a: string): string {
+  assertIsHexString(a)
+  a = ethjsUtil.stripHexPrefix(a)
+  return stripZeros(a) as string
+}
+
+/**
+ * Trims leading zeros from a `Buffer`, `String` or `Number[]`.
  * @param a (Buffer|Array|String)
  * @return (Buffer|Array|String)
  */
-export const unpad = function(a: any) {
-  a = ethjsUtil.stripHexPrefix(a)
+const stripZeros = function(a: any): Buffer | number[] | string {
   let first = a[0]
   while (a.length > 0 && first.toString() === '0') {
     a = a.slice(1)
     first = a[0]
   }
   return a
 }
-export const stripZeros = unpad
```

Comment:
I've removed stripZeros from the exports and made it the core logic of the unpad variants. Another approach might be to add `stripZerosBuffer`, `stripZerosHexString`, etc. Not certain what the right thing to do here is.

Reply:
Think this is a good solution.